File: 541 - Reverse String II.java
class Solution {
    public String reverseStr(String s, int k) {
        char[] c = s.toCharArray();
        int j = 0;
        for (int start = 0; start < s.length(); start += k * 2) {
            j = start + k - 1;
            this.swap(c, start, j);
        }
        return new String(c);
    }
    public void swap(char[] a, int i, int j) {
        j = Math.min(j, a.length - 1);
        char t;
        while (i < j) {
            t = a[i];
            a[i++] = a[j];
            a[j--] = t;
        }

    }
}

File: 541 - Reverse String II.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var reverseStr = function (s, k) {
    let t = s.split('');
    for (let i = 0; i < s.length; i += 2 * k) {
        swap(t, i, i + k - 1)
    }
    return t.join('');
};
function swap(s, i, j) {
    j = Math.min(j, s.length - 1);
    let t;
    while (i < j) {
        t = s[i];
        s[i++] = s[j];
        s[j--] = t;
    }
}

console.log(reverseStr('nmslijk', 2));